app/__init__.py

```diff
@@ -0,0 +1,374 @@
+                                        'date_created': item.date_created,
+                                        'date_modified': item.date_modified,
+                                        'creator_id': item.creator_id
+                                    }
+                                    buckets.append(a_bucket_object)
+                                response = jsonify(buckets)
+                                return make_response(response), 200
+                            else:
+                                response = jsonify({
+                                    'message': 'Zero returns no buckets'
+                                })
+                                return make_response(response), 202
+                        else:
+                            list_of_bucketlist = BucketList.query.filter_by(creator_id=user_id)
+                            bucketlist_objects_list = []
+
+                            for item in list_of_bucketlist:
+                                a_bucket_object = {
+                                                    'id': item.id,
+                                                    'name': item.name,
+                                                    'date_created': item.date_created,
+                                                    'date_modified': item.date_modified,
+                                                    'creator_id': item.creator_id
+                                }
+                                bucketlist_objects_list.append(a_bucket_object)
+                            #converting the bucket list objec into JSON
+                            response = jsonify(bucketlist_objects_list)
+                            return make_response(response), 200
+                else:
+                    response = {
+                        'message': 'User id is a string',
+                        'userid': user_id
+                    }
+                    return make_response(jsonify(response)), 401
+            else:
+                response = ({
+                    'message': 'The authorisation header should have a space between the Bearer and the token'
+                })
+                return make_response(response), 400
+        else:
+            response = jsonify({
+                'message': 'No authorisation header given'
+            })
+            return make_response(response), 401
+            
+    @app.route('/bucketlists/<int:id>', methods=['GET', 'PUT', 'DELETE'])
+    def bucketlist_manipulation(id, **kwargs):
+        '''a method that accepts a variable on its route and converts it into an integer
+        which is used in setting it as an id. The id allows editing and deleting a
+        particular object
+        '''
+        header = request.headers.get('Authorization')
+        splitted_header = header.split(' ')
+        token = splitted_header[1]
+        user_id = User.decode_token_to_sub(token)
```

Comment:
consider breaking off token decode process into a function to avoid repeating similar code